packages/core/src/connectivity/scp-cf/destination/destination-from-service.ts

```diff
@@ -320,43 +331,36 @@ class DestinationFromServiceRetriever {
     return this.fetchDestinationByToken(token);
   }
 
-  private async addOAuth2ClientCredentials(): Promise<Destination> {
+  /**
+   * The fetch by name delegates the oAuth token handling to the destination service.
+   * @private
+   */
+  private async fetchDestinationByClientCrendentialsGrant(): Promise<Destination> {
     const clientGrant = await serviceToken('destination', {
       userJwt: this.options.userJwt || this.providerClientCredentialsToken
     });
 
-    return fetchDestination(
-      this.destinationServiceCredentials.uri,
-      clientGrant,
-      this.name
-    );
+    return this.fetchDestinationByToken(clientGrant);
   }
 
-  private async addOAuthSamlAuth(
-    destination: Destination,
-    destinationOrigin: DestinationOrigin
-  ): Promise<Destination> {
-    /* This covers the two technical user propagation cases https://help.sap.com/viewer/cca91383641e40ffbe03bdc78f00f681/Cloud/en-US/3cb7b81115c44cf594e0e3631291af94.html
-   If the destination comes from the provider account the client credentials token from the xsuaa.url is used (provider token).
-   If the destination comes from the subscriber account the subscriber subdomain is used fetch the client credentials token (subscriber token).
-   */
-    if (destination.systemUser) {
+  private async fetchDestinationBySystemUser(
+    destinationResult: DestinationSearchResult
+  ): Promise<Destination | undefined> {
+    if (destinationResult.destination.systemUser) {
       const token =
-        destinationOrigin === 'provider'
+        destinationResult.origin === 'provider'
           ? this.providerClientCredentialsToken
           : await DestinationFromServiceRetriever.getSubscriberClientCredentialsToken(
               this.options
             );
       logger.debug(
-        `System user found on destination. The ${destinationOrigin} token: ${token} is used for destination fetching.`
+        `System user found on destination. The ${destinationResult.origin} token: ${token} is used for destination fetching.`
```

Comment:
Should we log the token? Maybe I am remembering wrong, but I think we wanted to remove that. If not, consider the readability. A long token in the middle of the sentence will make it hard to read the sentence. I would put it in the end with a colon and quotes `... token is used for destination fetching: "${token}".`

Reply:
Yes, we only log token in this case, and we decided to remove it, as it's useless and the "system user" related functionalities are deprecated from the BTP.